src/editor.rs

```diff
@@ -248,6 +248,7 @@ where
         Ok(())
     }
 
+    #[allow(clippy::unnecessary_wraps)]
```

Comment:
This seems like a weird clippy bug, it's passing for me locally when allowing this unecessary_wraps attribute.  As context there seems to be some open issues regarding clippy's strictness as a result it does not realise that a Result<_> is required for the process_keypress method in editor.rs

[here some similar open issues](https://github.com/rust-lang/rust-clippy/issues?q=is%3Aissue+is%3Aopen+unnecessary_wraps)